_docs/software_support_lifecycle.md

```diff
@@ -0,0 +1,80 @@
+
+1. A deprecation PR will be created no later that 3 months before an Operating System's End of Life date.
+2. Removal of the End of Life version from the metadata.json and tests to be completed by the End of Life date.
+3. Issues relating to the End of Life operating system will no longer be addressed after the End of Life date.
+4. Code allowing a module to be run on the End of Life operating system will be left in the module until one of the following milestones are reached after which time the code will be removed:
+    1. Maintaining the code prevents the module from moving forward.
+    2. 12 Months after the End of Life date.
+5. Pull Requests will continue to be accepted with the following exceptions:
+    1. Accepting such a pull request would be detrimental to the module's long term goals.
+    2. The code has already been removed as per the standards set in section 4.
+
+*maintainers include, but are not limited to, the community, company, or organization that maintains ownership over a specific piece of software or project.
+
+##### Support Table:
+
+| OS Name | Support Begins | End of Life | Notification PR | End of Support | Removal of Code |
+| ------- | -------------- | ----------- | --------------- | -------------- | --------------- |
+| CentOS/RHEL 6 | July 10th, 2011 | Nov 30th, 2020 | N/A | Nov 30th, 2020 | Nov 30th, 2021 |
+| CentOS/RHEL 7 | July 7th, 2014 | June 30th, 2024 | Mar 31st, 2024 | June 30th, 2024 | June 30th, 2025 |
+| CentOS/RHEL 8 | Sept 24th, 2019 | Dec 31, 2021 | Sept 30th, 2021 | Dec 31, 2021 | Dec 31, 2022 |
+| CentOS Stream | Jan 1st, 2022 | N/A | N/A | N/A | N/A |
+| Ubuntu 16.04 | Apr 21, 2016 | Apr 2021 | Jan 30th, 2021 | Apr 30th, 2021 | Apr 30th, 2022 |
+| Ubuntu 18.04 | Apr 26th, 2018 | Apr 2023 | Jan 30th, 2023 | Apr 30th, 2023 | Apr 30th, 2024 |
+| Ubuntu 20.04 | Apr 23rd, 2020 | Apr 2025 | Jan 30th, 2025 | Apr 30th, 2025 | Apr 30th, 2026 |
+| Debian 9 | June 17th, 2017 | June 2022 | Mar 31st, 2022 | June 30th, 2022 | June 30th, 2023 |
+| Debian 10 | July 6th, 2019 | 2024 | 2024 | 2024 | 2025 |
+| Fedora 31 | Oct 29th, 2019 | Nov 24th, 2020 | N/A | Nov 24th, 2020 | Nov 24th, 2021 |
+| Fedora 32 | Apr 28th, 2020 | May 2021 | Feb 2021 | May 31st, 2021 | May 31st, 2022 |
+| Fedora 33 | Oct 27th, 2020 | Nov 2021 | Aug 2021 | Nov 30th, 2021 | Nov 30th, 2022 |
+| OpenSUSE Leap 15.1 | May 22nd, 2019 | Jan 31st, 2021 | Dec 31st, 2020 | Jan 31st, 2021 | Jan 31st, 2022 |
+| OpenSUSE Leap 15.2 | July 2nd, 2020 | Dec 31st, 2021 | Sept 30th, 2021 | Dec 31st, 2021 | Dec 31st, 2022 |
+
+#### Puppet Versions
+
+Puppet support will fall in-line with the Puppet Open Source project's lifecycle.
+
+1. A Puppet version that has been labelled as End of Life by the Puppet Open Source project, will no longer be supported by Vox Pupuli.
+2. A deprecation PR will be created no later than 6 months before End of Life.
+3. All tests and data in metadata.json will be removed no later than the End of Life date for that version.
```

Comment:
i don't remember this specific outcome of our discussions
if this is happening, then we should branch a 5.x and allow the community to create PRs against that

Reply:
To avoid confusion it would be beneficial to outline the specifics so if people want to do it, there is a pattern. Like creating a branch named `puppet5`.

Reply:
There can be some leeway for `puppet5` due to the timing, but I am strongly against putting the pressure on an already over-strained Vox Pupuli community to support EoL software in the future.. 
It'd be different if we were a paid team offering Enterprise support, but we are not.

Reply:
Normally I'm inclined to agree. I certainly agree we're already over-strained. However, removing Puppet 5 code is typically more work than keeping it in. Except for the systemd daemon-reload fix in Puppet 6.1 I'm not aware of things that are really different between 5 and 6. That means it's typically only an entry in metadata.json that's different (which results in a different test matrix).